onnx/shape_inference/implementation.h

```diff
@@ -12,6 +12,37 @@
 namespace ONNX_NAMESPACE {
 namespace shape_inference {
 
+struct SymbolTable {
+  SymbolTable(GraphProto* g) : index_(0){
+    existing_symbols.clear();
+    AddExistingSymbolicDims(*g->mutable_input());
+    AddExistingSymbolicDims(*g->mutable_output());
+    AddExistingSymbolicDims(*g->mutable_value_info());
+  }
+  std::string createNew(std::string symbol_prefix="unk__") {
+    std::string newSymbol;
+    do {
+      newSymbol = symbol_prefix + std::to_string(index_++);
+    } while(existing_symbols.count(newSymbol) > 0);
+    return newSymbol;
+  };
+  private:
+    unsigned int index_;
+    std::unordered_set<std::string> existing_symbols;
+    void AddExistingSymbolicDims(google::protobuf::RepeatedPtrField<ValueInfoProto> protos) {
+      for (auto& proto : protos) {
+        auto tensorType = proto.mutable_type()->mutable_tensor_type();
```

Comment:
why mutable when you are not modifying the proto? you are simply collecting all the symbols right

Reply:
dont you need to check for sparse tensor and sequences?

Reply:
Good catch. I have added sparse_tensor and sequence for AddExistingSymbolicDims. Thank you!